cura/Settings/ExtruderStack.py

```diff
@@ -141,12 +141,11 @@ def getProperty(self, key: str, property_name: str, context: Optional[PropertyEv
             return result
 
         limit_to_extruder = super().getProperty(key, "limit_to_extruder", context)
-        if limit_to_extruder is not None:
-            if limit_to_extruder == -1:
-                limit_to_extruder = int(cura.CuraApplication.CuraApplication.getInstance().getMachineManager().defaultExtruderPosition)
+        if limit_to_extruder == -1:
+            limit_to_extruder = cura.CuraApplication.CuraApplication.getInstance().getMachineManager().defaultExtruderPosition
+        if limit_to_extruder:
             limit_to_extruder = str(limit_to_extruder)
-
-        if (limit_to_extruder is not None and limit_to_extruder != "-1") and self.getMetaDataEntry("position") != str(limit_to_extruder):
+        if limit_to_extruder != self.getMetaDataEntry("position"): 
```

Comment:
This case is entered when limit\_to\_extruder is None, which it did not before.

Also if the defaultExtruderPosition is -1 this case is also entered. I'm not sure how likely this is but it is best to code a bit defensively.